zipkin/src/main/java/zipkin/filter/FilterActivatedException.java

```diff
@@ -0,0 +1,14 @@
+package zipkin.filter;
+
+public class FilterActivatedException extends Throwable {
+  private final int httpCode;
+
+  public FilterActivatedException(String message, int httpCode) {
+    super(message);
+    this.httpCode = httpCode;
+  }
+
```

Comment:
this seems leaky abstraction as it isnt necessarily http but has http code. also extending throwable is a bit weird. maybe use a built-in exception or document how this is used as a signal. we don't currently define custom exceptions anywhere else in the codebase

Reply:
I'll look around from suitable built-ins. I'll add better documentation too. The reason I extended `Throwable` was because the error call takes a throwable: https://github.com/openzipkin/zipkin/pull/2072/files#diff-1b30356bd2b01dbbfd9152c17a1ac4baR148

In theory, I could copy what you did in that same function: `message.startsWith("Cannot store")`, but that isn't strongly typed and requires an implementor to have to be careful about their error strings. I wanted to avoid that drawback and make it easier on filter implementors. Happy to send another PR to clean replace `message.startsWith`.